In [2]:
import pandas as pd
import numpy as np

In [3]:
df=pd.read_csv(r"C:\Users\sagar\Downloads\property (1).csv")

In [4]:
df

Suburb           Address  Rooms Type      Price Method  \
0         Abbotsford      85 Turner St      2    h  1480000.0      S   
1         Abbotsford   25 Bloomburg St      2    h  1035000.0      S   
2         Abbotsford      5 Charles St      3    h  1465000.0     SP   
3         Abbotsford  40 Federation La      3    h   850000.0     PI   
4         Abbotsford       55a Park St      4    h  1600000.0     VB   
...              ...               ...    ...  ...        ...    ...   
13575  Wheelers Hill      12 Strada Cr      4    h  1245000.0      S   
13576   Williamstown     77 Merrett Dr      3    h  1031000.0     SP   
13577   Williamstown       83 Power St      3    h  1170000.0      S   
13578   Williamstown      96 Verdon St      4    h  2500000.0     PI   
13579     Yarraville        6 Agnes St      4    h  1285000.0     SP   

        SellerG        Date  Distance  Postcode  ...  Bathroom  Car  Landsize  \
0        Biggin   3/12/2016       2.5    3067.0  ...       1.0  1.0     202.0   
1        Biggin   4/02/2016       2.5    3067.0  ...       1.0  0.0     156.0   
2        Biggin   4/03/2017       2.5    3067.0  ...       2.0  0.0     134.0   
3        Biggin   4/03/2017       2.5    3067.0  ...       2.0  1.0      94.0   
4        Nelson   4/06/2016       2.5    3067.0  ...       1.0  2.0     120.0   
...         ...         ...       ...       ...  ...       ...  ...       ...   
13575     Barry  26/08/2017      16.7    3150.0  ...       2.0  2.0     652.0   
13576  Williams  26/08/2017       6.8    3016.0  ...       2.0  2.0     333.0   
13577     Raine  26/08/2017       6.8    3016.0  ...       2.0  4.0     436.0   
13578   Sweeney  26/08/2017       6.8    3016.0  ...       1.0  5.0     866.0   
13579   Village  26/08/2017       6.3    3013.0  ...       1.0  1.0     362.0   

       BuildingArea  YearBuilt  CouncilArea Lattitude  Longtitude  \
0               NaN        NaN        Yarra -37.79960   144.99840   
1              79.0     1900.0        Yarra -37.80790   144.99340   
2             150.0     1900.0        Yarra -37.80930   144.99440   
3               NaN        NaN        Yarra -37.79690   144.99690   
4             142.0     2014.0        Yarra -37.80720   144.99410   
...             ...        ...          ...       ...         ...   
13575           NaN     1981.0          NaN -37.90562   145.16761   
13576         133.0     1995.0          NaN -37.85927   144.87904   
13577           NaN     1997.0          NaN -37.85274   144.88738   
13578         157.0     1920.0          NaN -37.85908   144.89299   
13579         112.0     1920.0          NaN -37.81188   144.88449   

                       Regionname Propertycount  
0           Northern Metropolitan        4019.0  
1           Northern Metropolitan        4019.0  
2           Northern Metropolitan        4019.0  
3           Northern Metropolitan        4019.0  
4           Northern Metropolitan        4019.0  
...                           ...           ...  
13575  South-Eastern Metropolitan        7392.0  
13576        Western Metropolitan        6380.0  
13577        Western Metropolitan        6380.0  
13578        Western Metropolitan        6380.0  
13579        Western Metropolitan        6543.0  

[13580 rows x 21 columns]

#  1.)  For the suburb Altona, it is postulated that a typical property sells for 800,000. Use the data at hand to test this assumption. Is the typical property price really 800,000 or has it increased? Use a significance level of 5%. 

In [5]:
import pandas as pd
import numpy as np

# Filter Altona properties
altona_prices = df[df['Suburb'] == 'Altona']['Price']

# Hypothesized mean
mu = 800000

# Calculate sample statistics using pandas
n = altona_prices.count()
sample_mean = altona_prices.mean()
sample_std = altona_prices.std(ddof=1)  # Sample standard deviation

# Calculate the t-statistic using pandas calculations
t_stat = (sample_mean - mu) / (sample_std / np.sqrt(n))

# Calculate the two-tailed p-value using the t-distribution
from scipy.stats import t
p_value = 2 * (1 - t.cdf(abs(t_stat), df=n-1))

# Display the results
print("Sample Size:", n)
print("Sample Mean:", sample_mean)
print("Sample Standard Deviation:", sample_std)
print("T-Statistic:", t_stat)
print("P-Value:", p_value)


Sample Size: 74
Sample Mean: 834830.4054054054
Sample Standard Deviation: 291546.04547367844
T-Statistic: 1.0277020770199679
P-Value: 0.30748327130555486


The p-value (0.307) confirms that we fail to reject the null hypothesis. The typical property price in Altona is not significantly different from $800,000 at the 5% significance level. 

# 2 . For the year 2016, is there any difference in prices of properties sold in the summer months vs winter months? Consider months from October till March as winter months and the rest as summer months. Use a significance level of 5%. 

In [9]:
# Convert Date column to datetime
df['Date'] = pd.to_datetime(df['Date'], format='%d/%m/%Y')

# Filter properties sold in 2016
data_2016 = df[df['Date'].dt.year == 2016].copy()  # Use .copy() to avoid the warning

# Extract month from Date
data_2016['Month'] = data_2016['Date'].dt.month

# Define winter (Oct-Mar) and summer (Apr-Sep) months
winter_months = [10, 11, 12, 1, 2, 3]
summer_months = [4, 5, 6, 7, 8, 9]

# Split prices into winter and summer groups
winter_prices = data_2016[data_2016['Month'].isin(winter_months)]['Price']
summer_prices = data_2016[data_2016['Month'].isin(summer_months)]['Price']

# Perform two-sample t-test
from scipy.stats import ttest_ind
t_stat, p_value = ttest_ind(winter_prices, summer_prices, equal_var=False)  # Welch's t-test

# Display results
print("T-Statistic:", t_stat)
print("P-Value:", p_value)

# Check significance
if p_value < 0.05:
    print("Conclusion: Significant difference in prices (reject H0).")
else:
    print("Conclusion: No significant difference in prices (fail to reject H0).")

T-Statistic: 3.9211012778003753
P-Value: 8.950129747463378e-05
Conclusion: Significant difference in prices (reject H0).


# 3 .For the suburb of Abbotsford, what is the probability that out of 10 properties sold, 3 will not have car parking? Use the column car in the dataset. Round off your answer to 3 decimal places. 

In [10]:
from math import comb

# Filter properties in Abbotsford
abbotsford_data = df[df['Suburb'] == 'Abbotsford']

# Total properties in Abbotsford
total_properties_abbotsford = abbotsford_data.shape[0]

# Properties without car parking
properties_without_car = abbotsford_data[abbotsford_data['Car'] == 0].shape[0]

# Calculate p (probability of no car parking)
p_no_car = properties_without_car / total_properties_abbotsford

# Binomial calculation for P(X = 3) when n = 10 and k = 3
n = 10
k = 3
binomial_probability = comb(n, k) * (p_no_car**k) * ((1 - p_no_car)**(n - k))

# Round off to 3 decimal places
binomial_probability_rounded = round(binomial_probability, 3)

print("Probability of no car parking (p):", round(p_no_car, 3))
print("Probability of exactly 3 out of 10 properties without car parking:", binomial_probability_rounded)


Probability of no car parking (p): 0.268
Probability of exactly 3 out of 10 properties without car parking: 0.26


# In the suburb of Abbotsford, what are the chances of finding a property with 2 bathrooms? Round your answer to 3 decimal places. 

In [12]:
# Filter properties in Abbotsford
abbotsford_data = df[df['Suburb'] == 'Abbotsford']

# Total properties in Abbotsford
total_properties_abbotsford = abbotsford_data.shape[0]

# Properties with 3 rooms in Abbotsford
properties_with_3_rooms = abbotsford_data[abbotsford_data['Rooms'] == 3].shape[0]

# Calculate probability
probability = properties_with_3_rooms / total_properties_abbotsford

# Round to 3 decimal places
probability_rounded = round(probability, 3)

print("Total Properties in Abbotsford:", total_properties_abbotsford)
print("Properties with 3 Rooms:", properties_with_3_rooms)
print("Probability of 3 Rooms:", probability_rounded)


Total Properties in Abbotsford: 56
Properties with 3 Rooms: 20
Probability of 3 Rooms: 0.357


# In the suburb of Abbotsford, what are the chances of finding a property with 3 rooms? Round your answer to 3 decimal places. 

In [11]:
# Filter properties in Abbotsford
abbotsford_data = df[df['Suburb'] == 'Abbotsford']

# Total properties in Abbotsford
total_properties_abbotsford = abbotsford_data.shape[0]

# Properties with 2 bathrooms in Abbotsford
properties_with_2_bathrooms = abbotsford_data[abbotsford_data['Bathroom'] == 2].shape[0]

# Calculate probability
probability = properties_with_2_bathrooms / total_properties_abbotsford

# Round to 3 decimal places
probability_rounded = round(probability, 3)

print("Total Properties in Abbotsford:", total_properties_abbotsford)
print("Properties with 2 Bathrooms:", properties_with_2_bathrooms)
print("Probability of 2 Bathrooms:", probability_rounded)

Total Properties in Abbotsford: 56
Properties with 2 Bathrooms: 19
Probability of 2 Bathrooms: 0.339
